galsim/gsobject.py

```diff
@@ -674,7 +674,12 @@ def dilate(self, scale):
 
         @returns the dilated object.
         """
-        return self.expand(scale) * (1./scale**2)  # conserve flux
+        # equivalent to self.expand(scale) * (1./scale**2)
+        new_obj = galsim.Transform(self, jac=[scale, 0., 0., scale], flux_ratio=scale**-2)
+
+        if hasattr(self, 'noise'):
```

Comment:
Were we just kind of missing this all along?  Ugh.  If so, is there a unit test that would have caught this that we could add?